sympy/solvers/inequalities.py

```diff
@@ -1007,3 +1009,154 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+class UnboundedLinearProgrammingError(Exception):
+    pass
+
+
+class InfeasibleLinearProgrammingError(Exception):
+    pass
+
+
+def _pivot(M, i, j):
+    MM = Matrix.zeros(M.rows, M.cols)
+    for ii in range(M.rows):
+        for jj in range(M.cols):
+            if ii == i and jj == j:
+                MM[ii, jj] = 1 / M[i, j]
+            elif ii == i and jj != j:
+                MM[ii, jj] = M[ii, jj] / M[i, j]
+            elif ii != i and jj == j:
+                MM[ii, jj] = -M[ii, jj] / M[i, j]
+            else:
+                MM[ii, jj] = M[ii, jj] - M[ii, j] * M[i, jj] / M[i, j]
+    return MM
+
+
+def _simplex(M, R, S, random_seed=0):
+    """
+    Simplex method with randomized pivoting
+    http://web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
+    """
```

Comment:
Include a proper reference: title, author etc rather than just a URL. A URL is nice but if the URL stops working it should still be possible to identify the document that you are referring to.